libbpfgo/libbpfgo.go

```diff
@@ -598,14 +623,84 @@ func doAttachKprobeLegacy(prog *BPFProg, kp string, isKretprobe bool) (*BPFLink,
 var eventChannels = make(map[uintptr]chan []byte)
 var lostChannels = make(map[uintptr]chan uint64)
 
+func (m *Module) InitRingBuf(mapName string, eventsChan chan []byte) (*RingBuffer, error) {
+	bpfMap, err := m.GetMap(mapName)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := uintptr(bpfMap.fd)
+	if eventsChan == nil {
+		return nil, fmt.Errorf("events channel can not be nil")
+	}
+	eventChannels[ctx] = eventsChan
+
+	rb := C.init_ring_buf(bpfMap.fd)
+	if rb == nil {
+		return nil, fmt.Errorf("")
+	}
+
+	ringBuf := &RingBuffer{
+		rb:     rb,
+		bpfMap: bpfMap,
+		stop:   make(chan bool),
+	}
+	m.ringBufs = append(m.ringBufs, ringBuf)
+	return ringBuf, nil
+}
+
+func (rb *RingBuffer) Start() {
+	go rb.poll()
+}
+
+func (rb *RingBuffer) Stop() {
+	rb.stop <- true
+}
+
+func (rb *RingBuffer) Close() {
+	if rb.closed {
+		return
+	}
+	for {
+		if rb.stopped {
+			break
+		}
+	}
+	C.ring_buffer__free(rb.rb)
+	rb.closed = true
+}
+
+func (rb *RingBuffer) poll() error {
+
+	go func() {
+		<-rb.stop
+		rb.stopped = true
+	}()
+
+	for {
+		err := C.ring_buffer__poll(rb.rb, 300)
+		if err < 0 {
+			if syscall.Errno(-err) == syscall.EINTR {
+				continue
+			}
+			return fmt.Errorf("error polling ring buffer: %d", err)
+		}
+		
+		if rb.stopped {
```

Comment:
just a readability suggestion (totally optional) to move this stop condition to the for line
```
for !rb.stopped {
```